x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/accesscontrol/SecurityIndexReaderWrapperIntegrationTests.java

```diff
@@ -182,15 +173,12 @@ protected IndicesAccessControl getIndicesAccessControl() {
 
     public void testDLSWithLimitedPermissions() throws Exception {
         ShardId shardId = new ShardId("_index", "_na_", 0);
-        MapperService mapperService = mock(MapperService.class);
-        ScriptService  scriptService = mock(ScriptService.class);
-        when(mapperService.documentMapper()).thenReturn(null);
-        when(mapperService.simpleMatchToFullName(anyString()))
-                .then(invocationOnMock -> Collections.singletonList((String) invocationOnMock.getArguments()[0]));
-        when(mapperService.fieldType(Mockito.anyString())).then(invocation -> {
-            final String fieldName = (String) invocation.getArguments()[0];
-            return new KeywordFieldMapper.KeywordFieldType(fieldName);
-        });
+        MappingLookup mappingLookup = MappingLookupUtils.fromTypes(
+            new KeywordFieldType("field"),
+            new KeywordFieldType("f1"),
+            new KeywordFieldType("f2")
+        );
```

Comment:
these changes are great. They make what tests do much more readable, and also allow us to only plug in the needed behaviour. Before the field type lookup would always return the same field type regardless of the provided field name, which is not so easy to follow.